Geometry/GEMGeometryBuilder/plugins/ME0GeometryValidate.cc

```diff
@@ -0,0 +1,341 @@
+#include "Geometry/GEMGeometry/interface/ME0Layer.h"
+#include "Geometry/GEMGeometry/interface/ME0Chamber.h"
+#include "Geometry/CommonTopologies/interface/StripTopology.h"
+#include "Geometry/Records/interface/MuonGeometryRecord.h"
+
+#include "Fireworks/Core/interface/FWGeometry.h"
+
+#include "DataFormats/GeometrySurface/interface/RectangularPlaneBounds.h"
+#include "DataFormats/GeometrySurface/interface/TrapezoidalPlaneBounds.h"
+
+#include <TFile.h>
+#include <TH1.h>
+
+#include <limits>
+#include <string>
+#include <type_traits>
+#include <algorithm>
+
+using namespace std;
+
+template <class T>
+typename enable_if<!numeric_limits<T>::is_integer, bool>::type almost_equal(T x, T y, int ulp) {
+  // the machine epsilon has to be scaled to the magnitude of the values used
+  // and multiplied by the desired precision in ULPs (units in the last place)
+  return abs(x - y) <= numeric_limits<T>::epsilon() * abs(x + y) * ulp
+         // unless the result is subnormal
+         || abs(x - y) < numeric_limits<T>::min();
+}
+
+using namespace edm;
+
+class ME0GeometryValidate : public one::EDAnalyzer<> {
+public:
+  explicit ME0GeometryValidate(const ParameterSet&);
+  ~ME0GeometryValidate() override {}
+
+private:
+  void beginJob() override;
+  void analyze(const edm::Event&, const edm::EventSetup&) override;
+  void endJob() override;
+
+  void validateME0ChamberGeometry();
+  void validateME0EtaPartitionGeometry();
+  void validateME0EtaPartitionGeometry2();
+
+  void compareTransform(const GlobalPoint&, const TGeoMatrix*);
+  void compareShape(const GeomDet*, const float*);
+
+  float getDistance(const GlobalPoint&, const GlobalPoint&);
+  float getDiff(const float, const float);
+
+  void makeHistograms(const char*);
+  void makeHistograms2(const char*);
+  void makeHistogram(const string&, vector<float>&);
+
+  void clearData() {
+    globalDistances_.clear();
+    topWidths_.clear();
+    bottomWidths_.clear();
+    lengths_.clear();
+    thicknesses_.clear();
+  }
+
+  void clearData2() {
+    nstrips_.clear();
+    pitch_.clear();
+    stripslen_.clear();
+  }
+
+  edm::ESHandle<ME0Geometry> me0Geometry_;
+  FWGeometry fwGeometry_;
+  TFile* outFile_;
+  vector<float> globalDistances_;
+  vector<float> topWidths_;
+  vector<float> bottomWidths_;
+  vector<float> lengths_;
+  vector<float> thicknesses_;
+  vector<float> nstrips_;
+  vector<float> pitch_;
+  vector<float> stripslen_;
+  string infileName_;
+  string outfileName_;
+  int tolerance_;
+};
+
+ME0GeometryValidate::ME0GeometryValidate(const edm::ParameterSet& iConfig)
+    : infileName_(
+          iConfig.getUntrackedParameter<string>("infileName", "cmsRecoGeom-2026.root")),  //it was cmsGeom10.root
+      outfileName_(iConfig.getUntrackedParameter<string>("outfileName", "validateME0Geometry.root")),
+      tolerance_(iConfig.getUntrackedParameter<int>("tolerance", 6)) {
+  fwGeometry_.loadMap(infileName_.c_str());
+  outFile_ = TFile::Open(outfileName_.c_str(), "RECREATE");
```

Comment:
Should this eventually go in the Validation subsystem and use framework tools to manage ROOT output?

Reply:
@kpedro88 : the other muon geometries have similar/same analyzers in their packages, if for you it is not a big problem I'd prefer to leave it as it is.